src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -100,6 +101,43 @@ protected ImmutableArray<ISymbol> GetSymbols(IParameterSymbol parameter, int pos
                 .ToImmutableArray();
         }
 
+        private ImmutableArray<ITypeSymbol> SubstituteTypeParameters(ImmutableArray<ITypeSymbol> parameterTypeSymbols, SyntaxNode invocationExpression)
+        {
+            if (!parameterTypeSymbols.Any(t => t.IsKind(SymbolKind.TypeParameter)))
+            {
+                return parameterTypeSymbols;
+            }
+
+            var invocationSymbols = _context.SemanticModel.GetSymbolInfo(invocationExpression).GetAllSymbols();
```

Comment:
Not sure when this regressed, but this improves on the Visual Studio 2017 experience when there are multiple overloads involved.